Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/home/HomeFragmentViewModel.kt

```diff
@@ -195,25 +197,17 @@ class HomeFragmentViewModel @AssistedInject constructor(
         val pcrIdentifier = testPCR?.identifier ?: ""
         val ratIdentifier = testRAT?.identifier ?: ""
 
-        val positiveCoronaTests = listOfNotNull(testRAT, testPCR).filter { it.isPositive && it.isViewed }
-
         mutableListOf<HomeItem>().apply {
-            when {
-                // High risk is always visible regardless of test result
-                tracingItem is IncreasedRiskCard.Item -> add(tracingItem)
-
-                // No high risk card -> check corona test age against config duration
-                positiveCoronaTests.isNotEmpty() -> {
-                    if (positiveCoronaTests.all {
-                        it.shouldShowRiskCard(coronaTestParameters, timeStamper.nowUTC)
-                    }
-                    ) {
-                        add(tracingItem)
-                    } // else -> // Don't show risk card
-                }
 
-                // Otherwise show risk card
-                else -> add(tracingItem)
+            val currentRiskState = when (tracingItem) {
+                is IncreasedRiskCard.Item -> RiskState.INCREASED_RISK
+                is LowRiskCard.Item -> RiskState.LOW_RISK
+                is TracingFailedCard.Item -> RiskState.CALCULATION_FAILED
+                else -> null // tracing is disabled or calculation is currently in progress
```

Comment:
In that case it should show up , right ?

Reply:
I would not max the state and the test checks , please keep the logic separate, I see that you are passing `RiskState.LOW_RISK` all the time in the detector